### PR TITLE
Enable custom start commands and options to resolve GHA issues

### LINKED ISSUE #666: Github Action Deprecation: actions/upload-artifact@v3
> [!warning]
> ### Artifacts v3 brownouts
>Artifact actions v3 will be closing down by December 5, 2024. To raise awareness of the upcoming removal, we will temporarily fail jobs using v3 of actions/upload-artifact or actions/download-artifact. Builds that are scheduled to run during the brownout periods will fail.  The brownouts are scheduled for the following dates and times:
> – November 14, 12pm – 1pm EST
> – November 21, 9am – 5pm EST

_[Notice of Breaking Changes](https://github.blog/changelog/2024-11-05-notice-of-breaking-changes-for-github-actions/#artifacts-v3-brownouts) published Nov 5, 2024_

### This repository will be impacted by this brownout and deprecation, [based on this query.](https://github.com/search?q=org%3Aopensearch-project%20actions%2Fupload-artifact%40v3&type=code)

Want to avoid tracking this manually? Setup [dependabot](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates) to automatically bump version numbers of dependencies including GitHub Actions.

#### Add a .github/dependabot.yml configuration
```yaml
version: 2
updates:
  - package-ecosystem: "github-actions"
    directory: "/"
    schedule:
      interval: "weekly"
```
_[Example Source](https://github.com/opensearch-project/security/blob/main/.github/dependabot.yml#L11-L14)_

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Enable custom start commands and options to resolve GHA issues

### Related Issues
Closes #666 https://github.com/opensearch-project/asynchronous-search/pull/673
https://github.com/opensearch-project/opensearch-build/issues/5178

### Check List
- ~~[ ] New functionality includes testing.~~
- ~~[ ] New functionality has been documented.~~
- ~~[ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).~~
- [x] Commits are signed per the DCO using `--signoff`.
- ~~[ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).~~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/asynchronous-search/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
Hi @finnegancarroll @cwperks please take a look on this.

Thanks.

**Comment by amitgalitz** [comment]
Should we also update actions/setup-java to v4?

**Comment by peterzhuamazon** [comment]
> Should we also update actions/setup-java to v4?

This particular PR is just to update previously failed and unsupported upload artifact actions.
And yes you can definitely raise another PR for updating all actions if needed.

Thanks.

**Comment by amitgalitz** [comment]
> > Should we also update actions/setup-java to v4?
> 
> This particular PR is just to update previously failed and unsupported upload artifact actions. And yes you can definitely raise another PR for updating all actions if needed.
> 
> Thanks.

Sounds good, just mentioned cause I think I faced some node issue for the setup-java upgrades. Not a maintainer on this repo but updating on a different PR sounds good

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -30,16 +30,11 @@ jobs:
       # this image tag is subject to change as more dependencies and updates will arrive over time
       image: ${{ needs.Get-CI-Image-Tag.outputs.ci-image-version-linux }}
       # need to switch to root so that github actions can install runner binary on container without permission issues.
```
Comment (cwperks) [inline .github/workflows/build.yml:32]: nit: remove this line

**Review by cwperks — APPROVED** [review]
Thank you @peterzhuamazon! This PR looks good to me. I'm not a maintainer of this repo so a maintainer should also have a look.

**Pull request merged**